atorch/atorch/utils/loss_spike_utils.py

```diff
@@ -0,0 +1,157 @@
+""" loss spike utils
+Save loss spike to files;
+Decode loss spike and save Corresponding sample to a file. Using doc see ../docs/README-LOSS-SPIKE-UTIL.md
+"""
+import datetime
+import os
+
+import numpy as np
+
+from atorch.common.log_utils import default_logger as logger
+
+
+class LossSpikeBase:
+    def __init__(
+        self,
+        loss_spike_save_dir,
+        sample_data_paths,
+        each_sample_len,
+        min_iter,
+        min_loss,
+        loss_info_splitter="\t",
+        loss_sample_str_splitter=",",
+    ):
+        """
+        init params
+        Args:
+            loss_spike_save_dir: str, The directory where loss spike files are stored
+            sample_data_paths: The path information stored in the sample can be a user-defined structure,
+                such as name + path, tuple list:
+                [("wikipedia", "corpus/base"), ("zhihu", "/dataset/fd5061f6/data/tokenize_data/zhihu.lazy")]
+            each_sample_len: int, The length of a single sample
+            min_iter: int, Record the minimum iterations required for loss
+            min_loss: float, Record the minimum loss threshold required for loss
+            loss_info_splitter: str, Delimiter used to store loss spike information, default ='\t'
+            loss_sample_str_splitter: str, Delimiter used by str for a batch of loss or sample information, default =','
+        """
+        self.loss_spike_save_dir = loss_spike_save_dir
+        self.sample_data_paths = sample_data_paths
+        self.each_sample_len = each_sample_len
+        self.min_iter = min_iter
+        self.min_loss = min_loss
+        self.loss_info_splitter = loss_info_splitter
+        self.loss_sample_str_splitter = loss_sample_str_splitter
+
+        if not os.path.exists(loss_spike_save_dir):
+            raise ValueError("Param loss_spike_save_dir not exist!")
+        logger.info("Loss spike init success")
+
+    @staticmethod
+    def get_data_file_len(fpath, dtype):
+        with open(fpath) as f:
+            f.seek(0, 2)
+            return f.tell() // dtype.itemsize
+
+
+class TokenLossSpike(LossSpikeBase):
+    def save_loss(self, file_name, cur_loss, cur_iter, losses_str, sample_infos_str):
```

Comment:
What is the relationship between cur_loss and losses_str, or cur_iter and sample_infos_str?
What do losses_str and sample_infos_str mean in model training ?